mbed/mbed_terminal.py

```diff
@@ -0,0 +1,155 @@
+
+#!/usr/bin/env python2
+
+# Copyright (c) 2016 ARM Limited, All Rights Reserved
+# SPDX-License-Identifier: Apache-2.0
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied.
+
+
+# pylint: disable=too-many-arguments, too-many-locals, too-many-branches, too-many-lines, line-too-long,
+# pylint: disable=too-many-nested-blocks, too-many-public-methods, too-many-instance-attributes, too-many-statements
+# pylint: disable=invalid-name, missing-docstring, bad-continuation
+
+
+# Global class used for global config
+class MbedTerminal(object):
+    serial = None # Serial() object
+    port = None
+    baudrate = None
+    echo = None
+
+    def __init__(self, port, baudrate=9600, echo=True, timeout=10):
+        self.port = port
+        self.baudrate = int(baudrate)
+        self.timeout = int(timeout)
+        self.echo = False if str(echo).lower() == 'off' else True
+
+        try:
+            from serial import Serial, SerialException
+        except (IOError, ImportError, OSError):
+            return False
+
+        try:
+            self.serial = Serial(self.port, baudrate=self.baudrate, timeout=self.timeout)
+            self.serial.flush()
+            self.serial.reset_input_buffer()
+        except Exception as e:
+            print 'error'
+            self.serial = None
+            return False
+
+    def terminal(self, print_header=True):
+        try:
+            import serial.tools.miniterm as miniterm
+        except (IOError, ImportError, OSError):
+            return False
+
+        term = miniterm.Miniterm(self.serial, echo=self.echo)
+        term.exit_character = '\x03'
+        term.menu_character = '\x14'
+        term.set_rx_encoding('UTF-8')
+        term.set_tx_encoding('UTF-8')
+
+        def console_print(text):
+            term.console.write('--- %s ---\n' % text)
+
+        def get_print_help():
+            return """
+--- Mbed Serial Terminal (0.3a)
+--- Based on miniterm from pySerial
+---
+--- CTRL+B     Send Break (reset target)
```

Comment:
@screamerbg Are these just comments within the file? Can they be changed to match the lowercase commands in the README?